### DISCUSSION TITLE
Unloading of assemblies when exiting PIE

### DISCUSSION THREAD

**Discussion by dagillespie** [nxrighthere/UnrealCLR#110] [Help] [answered]
When exiting PIE mode when using external dependencies (eg as in https://github.com/nxrighthere/UnrealCLR/discussions/108), the application hangs and the following warning and error message appears:

`LogUnrealCLR: Warning: UnrealCLR::Module::Log: Unloading of assemblies took more time than expected. Trying to unload assemblies to the next breakpoint...
LogUnrealCLR: Error: UnrealCLR::Module::Log: Unloading of assemblies was failed! This might be caused by running threads, strong GC handles, or by other sources that prevent cooperative unloading.`

A simple example that causes this is referencing NewtonSoft. Is there a way to resolve this?

**Comment by nxrighthere** [comment, marked as answer]
If 3rd-party dependencies don't support unloadability with graceful unloading of resources, then developers should debug what causing this and work on that. More can be found [here](https://docs.microsoft.com/en-us/dotnet/standard/assembly/unloadability).
